cfme/tests/infrastructure/test_timelines.py

```diff
@@ -1,242 +1,181 @@
 
-@pytest.mark.meta(blockers=[BZ(1429962, forced_streams=["5.7"])])
-def test_infra_vm_diagnostic_timelines(gen_events, new_vm, mark_vm_as_appliance, appliance):
-    """Tests timelines on settings->diagnostics page
 
-    Metadata:
-        test_flag: timelines, provision
+class VMEvent(object):
+    """Class for generating  events on a VM in order to check it on Timelines.
+    Args:
+        vm: A VM object (Object)
     """
-    # go diagnostic timelines
-    wait_for(count_events, [appliance.server, new_vm], timeout='5m',
-             fail_condition=0, message="events to appear")
-
-
-class TestInfraVmEventRESTAPI(object):
-    @pytest.fixture(scope="module")
-    def a_provider(self, request):
-        return _a_provider(request)
-
-    @pytest.fixture(scope="module")
-    def vm(self, request, a_provider, appliance):
-        return _vm(request, a_provider, appliance.rest_api)
-
-    @pytest.mark.parametrize("from_detail", [True, False], ids=["from_detail", "from_collection"])
-    def test_vm_add_event(self, vm, from_detail, appliance):
-        """Test that checks whether adding a event using the REST API works.
-        Prerequisities:
-            * A VM
-        Steps:
-            * Find the VM's id
-            * Prepare the event data
-            * Call either:
-                * POST /api/vms/<id> (method ``add_event``) <- the event data
-                * POST /api/vms (method ``add_lifecycle_event``) <- the event data
-                    and resources field specifying list of dicts containing hrefs to the VMs,
-                    in this case only one.
-            * Verify that appliance's database contains such entries in table ``event_streams``
-        Metadata:
-            test_flag: rest
+    ACTIONS = {
+        'create': {
+            'tl_event': ('VmDeployedEvent', 'USER_RUN_VM'),
+            'tl_category': 'Creation/Addition',
+            'db_event_type': ('vm_create', 'USER_RUN_VM'),
+            'emit_cmd': '_setup_vm'
+        },
+        'start': {
+            'tl_event': ('VmPoweredOnEvent', 'USER_STARTED_VM'),
+            'tl_category': 'Power Activity',
+            'db_event_type': 'vm_start',
+            'emit_cmd': '_power_on'
+        },
+        'stop': {
+            'tl_event': ('VmPoweredOffEvent', 'USER_STOP_VM'),
+            'tl_category': 'Power Activity',
+            'db_event_type': 'vm_poweroff',
+            'emit_cmd': '_power_off'
+        },
+        'suspend': {
+            'tl_event': ('VmSuspendedEvent', 'USER_SUSPEND_VM'),
+            'tl_category': 'Power Activity',
+            'db_event_type': 'vm_suspend',
+            'emit_cmd': '_suspend'
+        },
+        'rename': {
+            'tl_event': 'VmReconfiguredEvent',
+            'tl_category': 'Configuration/Reconfiguration',
+            'db_event_type': 'VmRenamedEvent',
+            'emit_cmd': '_rename_vm'
+        },
+    }
+
+    def __init__(self, vm, event):
+        self.vm = vm
+        self.event = event
```

Comment:
REQUIRED: please replace this with something like
```
self.event = event
self.__dict__.update(self.ACTIONS[self.event])
```
I doubt this is good idea to use self.ACTION[blabla] in every method.